cli/src/pcluster/validators/iam_validators.py

```diff
@@ -57,6 +59,33 @@ def _validate(self, policy: str):
             self._add_failure(str(e), FailureLevel.ERROR)
 
 
+class IamResourcePrefixValidator(Validator):
+    """
+    Iam Resource Prefix Validator.
+
+    Verify if the Resource Prefix belongs in the permission boundary if given.
+    """
+
+    IAM_PATH_NAME_REGEX = r"^((/[a-zA-Z0-9_.,+@=-]+)+)/"
+    IAM_ROLE_NAME_PREFIX = r"^[a-zA-Z0-9_.,+@=-]+$"
+
+    def _validate(self, resource_prefix: str):
+        try:
+            if not (
+                re.match(IamResourcePrefixValidator.IAM_PATH_NAME_REGEX, resource_prefix)
+                or re.match(IamResourcePrefixValidator.IAM_ROLE_NAME_PREFIX, resource_prefix)
+            ):
+                self._add_failure(
+                    f"Resource Prefix {resource_prefix} provided does not fall under the accepted pattern",
```

Comment:
Is the error message approved by PM?
I'd rephrase with something like "Unsupported format for ResourcePrefix {resource_prefix} " or "ResourcePrefix {resource_prefix} does not comply with expected format."

Furthermore, I'd provide some suggestions to our customers about the expected format.
For example, we could suggest them to have a look at IAM documentation

Reply:
No the message wasnt approved. I will take your suggestion to change it.

Reply:
> I was thinking of maybe later adding the link to our documentation where we mention the exact requirements.

I agree. Let's put a link to our public docs https://docs.aws.amazon.com/parallelcluster/latest/ug/Iam-v3.html

Reply:
I propose the message: "ResourcePrefix {resource_prefix} is not valid. Please, refer to our public docs for further details."

Reply:
Changed the Failure message as requested.